packages/core/src/App/Containers/RealAccountSignup/account-wizard.jsx

```diff
@@ -291,6 +291,10 @@ const AccountWizard = observer(props => {
         delete clone?.confirmation_checkbox;
         delete clone?.crs_confirmation;
 
+        if (modifiedProps?.account_settings?.country_code === 'es' && clone?.resident_self_declaration)
```

Comment:
I suggest not to hard-code such business logic in the code even in v1.

Reply:
moved the logic to hook